### PR TITLE
Split install and configure commands

### LINKED ISSUE #34: Split install command into install and configure
`install`:
Installs the post-save hook in the jupyter configuration.

 - log that jupyter servers need to be restarted if installed for first time or new version

`configure`:
Creates the `.nbautoexport` configuration file in the 

 - error if not `install` has not been run

Other potential related fixes:
 - Rename "save progress indicator" and "sentinel" to be configuration throughout

### PR DESCRIPTION
**Pull request by jayqi**
**Splitting `install` command**

- Split `install` into two commands:
  - `install` installs the post-save hook initialization in the Jupyter config
  - `configure` creates the `.nbautoexport` configuration file in a directory
- When running the `configure` command, if the post-save hook initialization is absent or out-of-date, there will be a warning printed.
- Add a new `--jupyter-config` option to `install` if user wants to install the post-save hook initialization to a Jupyter config in a nonstandard location. 

**Other**

- Found a bug from #30 where new `clean` configuration option was not being passed through `install_sentinel`. Fixed by changing the interface of `install_sentinel` to accept a config object instead specifying each option. This reduces likelihood of similar future bugs by reducing the number of places that reproduce the `NbAutoexportConfig` interface. 
- Replace `pkg_resources.parse_version` with `packaging.version.parse`. This is what `pkg_resources` [calls under the hood](https://github.com/pypa/setuptools/blob/aff64ae89e00e25fb3868bf528a14c18e7af0cf4/pkg_resources/__init__.py#L127-L131) and is a lighter-weight dependency. 
- Downside of really high test coverage is then it becomes a pain to strictly maintain. Add a `.codecov` configuration to:
  - Allow up to a 2% decrease in coverage against base
  - Remove the `reach` graph, which I never found that interpretable but is really visually obnoxious. 

Closes #34. I did not unify the naming and opened #37 separately to avoid adding noise to this PR's diff.

### CODE REVIEW

File: nbautoexport/nbautoexport.py [hunk 1]
```diff
@@ -44,10 +46,18 @@ def main(
         help="Show nbautoexport version.",
     ),
 ):
-    """Exports Jupyter notebooks to various file formats (.py, .html, and more) upon save,
-    automatically.
+    """Automatically export Jupyter notebooks to various file formats (.py, .html, and more) upon
```
Comment (r-b-g-b) [inline nbautoexport/nbautoexport.py:49]: Ah just noticing that `typer` very nicely adjusts the width of these docstrings when they are printed out in the command line. Now the docstrings can be written with column width of 100 in the source code, but printed out to column width of 80 for the 0.1% of ppl who use ancient monitors.

File: tests/test_cli_configure.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+def test_install_no_jupyter_config_warning(tmp_path, monkeypatch):
+    def mock_jupyter_config_dir():
+        return str(tmp_path)
+
+    monkeypatch.setattr(jupyter_config, "jupyter_config_dir", mock_jupyter_config_dir)
+
+    result = CliRunner().invoke(app, ["configure", str(tmp_path)])
+    assert result.exit_code == 0
+    assert "Warning: nbautoexport is not properly installed with Jupyter." in result.output
```
Comment (r-b-g-b) [inline tests/test_cli_configure.py:108]: This test and `test_install_no_initialize_warning` are failing. This test should try to read a non-existent config from the `tmp_path` but instead it appears to be reading from my actual Jupyter config file (which contains a pre-versioning version of the post-save hook). I suppose the `mock_jupyter_config_dir` does not mock the `jupyter_config_dir()` call that happens _within_ the CLI call.

```
_____________________________________________ test_install_no_jupyter_config_warning _____________________________________________
                                                                                                                                  
tmp_path = PosixPath('/tmp/pytest-of-robert/pytest-0/test_install_no_jupyter_config0')                                            
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f678e302550>                                                          
                                
    def test_install_no_jupyter_config_warning(tmp_path, monkeypatch):
        def mock_jupyter_config_dir():     
            return str(tmp_path)                                                                                                  
                                                                                                                                  
        monkeypatch.setattr(jupyter_config, "jupyter_config_dir", mock_jupyter_config_dir)
                  
        result = CliRunner().invoke(app, ["configure", str(tmp_path)])                                                           
        assert result.exit_code == 0
>       assert "Warning: nbautoexport is not properly installed with Jupyter." in result.output
E       assert 'Warning: nbautoexport is not properly installed with Jupyter.' in "Warning: nbautoexport initialize is an older ve
rsion. Please run 'install' command to update.\n"
E        +  where "Warning: nbautoexport initialize is an older version. Please run 'install' command to update.\n" = <Result okay
>.output                                                                                                                          

tests/test_cli_configure.py:108: AssertionError                                                                                   
_______________________________________________ test_install_no_initialize_warning _______________________________________________
                                
tmp_path = PosixPath('/tmp/pytest-of-robert/pytest-0/test_install_no_initialize_war0')
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f678e2f6e80>                  
                                
    def test_install_no_initialize_warning(tmp_path, monkeypatch):    
        def mock_jupyter_config_dir():
            return str(tmp_path)                                                                                                  
                                                                                                                                          monkeypatch.setattr(jupyter_config, "jupyter_config_dir", mock_jupyter_config_dir)
                                                                                                                                          (tmp_path / "jupyter_notebook_config.py").touch()
     
        result = CliRunner().invoke(app, ["configure", str(tmp_path)])
        assert result.exit_code == 0                                                                                              
>       assert "Warning: nbautoexport is not properly installed with Jupyter." in result.output
E       assert 'Warning: nbautoexport is not properly installed with Jupyter.' in "Warning: nbautoexport initialize is an older ve
rsion. Please run 'install' command to update.\n"                                                                                 
E        +  where "Warning: nbautoexport initialize is an older version. Please run 'install' command to update.\n" = <Result okay
>.output      
                                                                                                                                  
tests/test_cli_configure.py:121: AssertionError
```

Comment (jayqi) [reply tests/test_cli_configure.py:105]: I'm a little surprised this happened given that the other tests seemed to pass, but I'm trying a different approach which I am confident should work.

File: nbautoexport/nbautoexport.py (hunk 1 above)
Comment (jayqi) [reply nbautoexport/nbautoexport.py:49]: Yeah, this actually annoyed me recently for a different project where I wanted to put some ASCII art in my help string but `typer` totally destroyed it with formatting. 😂 😢

**Comment by jayqi** [comment]
@r-b-g-b I think the issue you had should be fixed. Ready for another look.

File: tests/test_cli_configure.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    assert result.exit_code == 0
+    with (tmp_path / ".nbautoexport").open("r") as fp:
+        config = json.load(fp)
+
+    expected_config = NbAutoexportConfig(export_formats=["script", "html"], organize_by="notebook")
+    assert config == expected_config
+
+
+def test_configure_no_jupyter_config_warning(tmp_path, monkeypatch):
+    monkeypatch.setenv("JUPYTER_CONFIG_DIR", str(tmp_path))
```
Comment (r-b-g-b) [inline tests/test_cli_configure.py:101]: :+1:

**Pull request merged**